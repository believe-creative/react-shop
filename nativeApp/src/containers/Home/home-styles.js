import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  home: {
    backgroundColor: '#eeeeee',
  },
	banner: {
		backgroundColor: '#6c6c6c',
		paddingTop: 25,
		paddingBottom: 25,
		paddingLeft: 50,
		paddingRight: 50,
		flex:1,
		justifyContent: 'flex-start',
		alignItems: 'flex-start',
	},
	img: {
		width: null, 
		height: 300, 
		flex:1,
	},
	h1: {
		fontSize: 36,
		fontWeight: '700',
		color: '#fff',
		marginBottom: 10, 
	},
	h2: {
		fontSize: 22,
		fontWeight: '700',
		color: '#fff', 
		marginBottom: 10,
	},
	black: {		
		color: '#2e2e2e', 
	},
	red: {
		color: '#f62f5e',
	},
	white: {
		color: '#fff',
	},
	button: {
		backgroundColor: '#f62f5e',
		fontSize:18,
		fontWeight: '700',
		color: '#fff',
		borderRadius: 20,
		paddingTop: 15,
		paddingBottom: 15,
		paddingLeft: 30,
		paddingRight: 30,
		textAlign: 'center',
		minWidth: 150, 
	},
	shop_now_panel:{
		margin:15,
		marginBottom: 30,
		padding: 25,
		backgroundColor: '#fff',
		position: 'relative',
		shadowColor: '#000',
  		shadowOffset: {width: 0, height: 2},
  		shadowOpacity: 0.8,
  		shadowRadius: 2,
		elevation: 5,
	},
	product_panel:{
		flex:1,
		alignItems: 'center',
		justifyContent: 'center',
		flexDirection: 'column',
	},
	sale:{
		backgroundColor: '#00d3ca',
		color:'#fff',
		paddingTop: 5,
		paddingBottom: 5,
		paddingLeft: 10,
		paddingRight: 10,
		width: 'auto',
		position: 'absolute',
		left: 0,
		top: 0,
		zIndex: 1,
		fontSize: 16,
		fontWeight: '400',
		},
	shop_now: {
		textAlign: 'center',
		display:'flex',
		flex:1,
		alignItems: 'center',
		justifyContent: 'center',
	},
	shoptxt: {
		marginBottom:15,
		textAlign:'center',
		fontSize: 16,
		fontWeight: '400',
	},
	wow_block: {
		backgroundColor: '#00d3ca',
		padding: 15,
		margin: 15,
		marginBottom: 30, 
		minHeight: 345,
		flex:1,
		flexDirection: 'column',
		alignItems: 'center',
		justifyContent: 'center',
	},
	game_begin_img_block: {
		marginLeft: 15,
		marginRight: 15,
		marginTop: 15,
	},
	game_sub_block: {
		marginLeft:15,
		marginRight: 15,
		marginTop: 0,
		marginBottom: 40,
		padding: 30,
		backgroundColor: '#fff',
		position: 'relative',
		shadowColor: '#000',
  		shadowOffset: {width: 0, height: 2},
  		shadowOpacity: 0.8,
  		shadowRadius: 2,
		elevation: 5,
	},
	center: {
		textAlign: 'center',
	},
	
	/* product page */
	
	body: {
		backgroundColor: '#eeeeee',
		paddingTop: 30,
		paddingBottom: 30,
	},
	item: {
		backgroundColor: '#fff',
		padding: 25,
		margin: 15,
		paddingBottom: 40,
		position: 'relative',
		textAlign: 'center',
		marginBottom: 30,
		flex:1,
		alignItems: 'center',
		justifyContent: 'center',
		shadowColor: '#000',
  		shadowOffset: {width: 0, height: 10},
  		shadowOpacity: 0.8,
		shadowRadius: 2,
		elevation: 6, 
	},
	price: {
		fontWeight: 'bold',
		fontSize: 18,
		marginBottom: 15,
	},
	
	/* Topblock styles */
	
	topbar:{
		backgroundColor: '#f7f7f7',
		paddingTop:10,
		paddingBottom: 10,
		paddingLeft: 15,
		paddingRight:15,
		flexDirection: 'row',
		justifyContent:'space-between',
		alignItems:'center',
	},
	topleft:{
		flex:1,
		flexDirection: 'row',
	},
	toptext:{
		fontSize:15,
		fontWeight: '700',
		color:'#000',
	},
	signin:{
		fontSize:15,
		fontWeight: '700',
		color:'#f62f5e',
		marginLeft: 10,
	},	
	
});