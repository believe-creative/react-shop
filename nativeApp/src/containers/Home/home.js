/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import Footer from '../../components/Footer/footer';
import {Platform, StyleSheet, Text, View,Button, Linking, TouchableOpacity, Image, ScrollView} from 'react-native';
import { connect } from "react-redux";
import * as Actions from "../../actions";
import { getCookie } from "../../services/helpers";
import { createStackNavigator, createAppContainer } from 'react-navigation';
import Immutable from "immutable";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import createReducers from "../../reducers";
import rootSaga from "../../sagas";
import {store} from "../../store";
import {styles} from './home-styles'; 

type Props = {};

function demoAsyncCall() {
  return new Promise(resolve => setTimeout(() => resolve(), 2500));
}

class HomeScreen extends Component {
  constructor(props) {
    super(props);
    this.state = { token: "null", loaded: true, loading: true };

  }
  componentDidMount() {

  

  }

  componentWillReceiveProps(props) {
    if (props.token) {
      if (this.state.token !== props.token) {
        this.setState({ token: props.token });
        if (!this.props.categories)
          this.props.getCategories({ token: props.token });
      }
    }

  }
  render() {      
      return (        
			<ScrollView style={styles.home}>               
          <View style={styles.banner}>
			 	<Text style={styles.h1}>Background and development</Text>
			 	<Text style={styles.h2}>Convergent the dictates of the consumer: background and development</Text>
          	<TouchableOpacity onPress={() => Linking.openURL('/categories')}><Text style={styles.button}>View All</Text></TouchableOpacity>
			 </View> 
            <View style={styles.shop_now_panel}>
            <View style={styles.product_panel}>
					<Text style={styles.sale}>Sale</Text>
              <View style={styles.shop_now}>
					  <Image style={{width: 100, height: 100, marginBottom: 10}}                 
                  source={require('../../images/bag.png')}                  
                />
                <Text style={{...styles.h2, ...styles.title1}}>Vera Bradley</Text>  
                <Text style={styles.shoptxt}>
                  Carry the day in the style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring colorful faux leather trim,this tote offers a roomy interior plus just enough perfectly placed. 
                </Text>                
                <TouchableOpacity onPress={() => Linking.openURL('/categories')}><Text style={styles.button}>Shop Now</Text></TouchableOpacity>                 
              </View>
            </View>
          </View>
			 
			  <View style={styles.wow_block}>
                  <Text h1>WOW</Text>
                  <Text h2 style={{"color":"red"}}>Check</Text>                  
                  <Text h2 style={{"color":"red"}}>WHAT!</Text>
           </View>			 
				<View style={styles.wow_block}>
            	<Text h1>Men</Text>
            </View>
          <View style={styles.register_panel}>
            <View style={styles.shop_now_panel}>              
              <View>
                <View style={styles.game_begin_block}>
                <Image
                  style={{width: 50, height: 50}}                 
                  source={require('../../images/pop_image.png')}                  
                />
                  <View style={styles.game_sub_block}>
                    <Text h1>Let The Game begin</Text>
                    <Text h2>Registration is on - get ready for the Open</Text>
                    <TouchableOpacity onPress={() => Linking.openURL('/login')}>
                      <Text style={styles.button}>
                        Register
                      </Text>
                    </TouchableOpacity> 
                  </View>
                </View>
              </View>
            </View>
          </View>
          <Footer />           
      </ScrollView>
        );
      }
}

const productRequest = Actions.products.request;
const checkUserLogin = Actions.checkUserLogin.request;

const mapStateToProps = state => ({
  categories: state.get("products").categories,
  cart: state.get("products").cart,
  token: state.get("user").token
});

const mapDispatchToProps = dispatch => ({
  productRequest,
  checkUserLogin,
  getCategories: data => dispatch(Actions.getCategories.request(data)),
  getCartProducts: inCartId =>
    dispatch(Actions.getCartProducts.request(inCartId)),
  getToken: () => dispatch(Actions.getToken.request())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeScreen);
