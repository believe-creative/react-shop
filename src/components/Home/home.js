import React, { Component } from "react";
import Container from "react-bootstrap/Container";
import "../../scss/home.scss";

export default class Home extends Component {
  render() {
    return (
      <div className="home">
        <Container>
          <h1>Home Page</h1>
        </Container>
      </div>
    );
  }
}
