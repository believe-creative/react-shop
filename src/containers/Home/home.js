import React, { Component } from "react";
import { Switch, Route } from "react-router";
import { LinkContainer } from "react-router-bootstrap";
import Container from "react-bootstrap/Container";
import bag from "../../images/bag.png";
import pop_image from "../../images/pop_image.png";
import "../../scss/home.scss";

export default class Home extends Component {
  render() {
    return (
      <div className="home">
        <div className="row header_panel">
          <Container>
            <div className="col-md-6">
              <h1>Background and development</h1>
              <h2>
                Convergent the dictates of the consumer: background and
                development
              </h2>
              <p>
                <LinkContainer to="/categories" className="btn btn-lg">
                  <a>View All</a>
                </LinkContainer>
              </p>
            </div>
          </Container>
        </div>
        <Container>
          <div className="shop_now_panel">
            <div className="row product_panel">
              <div className="col-md-4 product_img">
                <div className="sale">SALE</div>
                <img src={bag} />
              </div>
              <div className="col-md-8 shop_now">
                <h2>Vera Bradley</h2>
                <p>
                  Carry the day in the style with this extra-large tote crafted
                  in our chic B.B. Collection textured PVC. Featuring colorful
                  faux leather trim,this tote offers a roomy interior plus just
                  enough perfectly placed.
                </p>
                <p>
                  <a href="/checkout" className="btn btn-lg">
                    Shop Now
                  </a>
                </p>
              </div>
            </div>
          </div>
          <div className="register_panel">
            <div className="row shop_now_panel">
              <div className="col-md-4">
                <div className="wow_block">
                  <h1>WOW</h1>
                  <h2 className="red">Check</h2>
                  <h2 className="red">WHAT!</h2>
                </div>
                <div className="wow_men_block">
                  <h1>Men</h1>
                </div>
              </div>
              <div className="col-md-8 ">
                <div className="game_begin_block">
                  <div className="pop">POP</div>
                  <img src={pop_image} />
                  <div className="game_sub_block">
                    <h1>Let The Game begin</h1>
                    <h2>Registration is on - get ready for the Open</h2>
                    <p>
                      <a href="#" className="btn btn-lg">
                        Register
                      </a>
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </Container>
        {/* <div className="subscribe_panel">
          <div className="row">
            <div className="col-md-6 offset-md-3 subscribe_block">
              <h2 className="red">10% Discount for your subscription</h2>
              <h3>
                Carry the day in style wish this extra-large tote crafted in our
                chic B.B Collection textured PVC. Featuring colorful faux
                leather trim,this tote offers a roomy interior
              </h3>
              <div className="input_search">
                {" "}
                <form action="#" method="post">
                  {" "}
                  <i className="fa fa-envelope" />
                  <input
                    type="text"
                    placeholder="       your email here"
                    className="search"
                  />
                  <a href="#" className="btn btn-md">
                    Subscribe
                  </a>
                </form>
              </div>
            </div>
          </div>
        </div> */}
      </div>
    );
  }
}
