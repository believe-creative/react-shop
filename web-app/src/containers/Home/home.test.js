import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Home from './home';
describe("MyComponent", () => {
  it("should render my component", () => {
    const wrapper = shallow(<Home />);
    // console.log("wrapper",wrapper.node.props);
    expect(wrapper).toMatchSnapshot();

  });
});
