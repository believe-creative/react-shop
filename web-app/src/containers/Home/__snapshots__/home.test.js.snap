// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render my component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="home"
  >
    <div
      className="row header_panel"
    >
      <ForwardRef(Bootstrap(Container))>
        <div
          className="col-md-6"
        >
          <h1>
            Background and development
          </h1>
          <h2>
            Convergent the dictates of the consumer: background and development
          </h2>
          <p>
            <LinkContainer
              activeClassName="active"
              className="btn btn-lg"
              exact={false}
              replace={false}
              strict={false}
              to="/categories"
            >
              <a
                href="/"
              >
                View All
              </a>
            </LinkContainer>
          </p>
        </div>
      </ForwardRef(Bootstrap(Container))>
    </div>
    <ForwardRef(Bootstrap(Container))>
      <div
        className="shop_now_panel"
      >
        <div
          className="row product_panel"
        >
          <div
            className="col-md-4 product_img"
          >
            <div
              className="sale"
            >
              SALE
            </div>
            <img
              alt="bag.png"
              src="bag.png"
            />
          </div>
          <div
            className="col-md-8 shop_now"
          >
            <h2>
              Vera Bradley
            </h2>
            <p>
              Carry the day in the style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring colorful faux leather trim,this tote offers a roomy interior plus just enough perfectly placed.
            </p>
            <p>
              <a
                className="btn btn-lg"
                href="/categories"
              >
                Shop Now
              </a>
            </p>
          </div>
        </div>
      </div>
      <div
        className="register_panel"
      >
        <div
          className="row shop_now_panel"
        >
          <div
            className="col-md-4"
          >
            <div
              className="wow_block"
            >
              <h1>
                WOW
              </h1>
              <h2
                className="red"
              >
                Check
              </h2>
              <h2
                className="red"
              >
                WHAT!
              </h2>
            </div>
            <div
              className="wow_men_block"
            >
              <h1>
                Men
              </h1>
            </div>
          </div>
          <div
            className="col-md-8 "
          >
            <div
              className="game_begin_block"
            >
              <img
                alt="pop_image.png"
                src="pop_image.png"
              />
              <div
                className="game_sub_block"
              >
                <h1>
                  Let The Game begin
                </h1>
                <h2>
                  Registration is on - get ready for the Open
                </h2>
                <p>
                  <a
                    className="btn btn-lg"
                    href="/login"
                  >
                    Register
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </ForwardRef(Bootstrap(Container))>
  </div>,
  "nodes": Array [
    <div
      className="home"
    >
      <div
        className="row header_panel"
      >
        <ForwardRef(Bootstrap(Container))>
          <div
            className="col-md-6"
          >
            <h1>
              Background and development
            </h1>
            <h2>
              Convergent the dictates of the consumer: background and development
            </h2>
            <p>
              <LinkContainer
                activeClassName="active"
                className="btn btn-lg"
                exact={false}
                replace={false}
                strict={false}
                to="/categories"
              >
                <a
                  href="/"
                >
                  View All
                </a>
              </LinkContainer>
            </p>
          </div>
        </ForwardRef(Bootstrap(Container))>
      </div>
      <ForwardRef(Bootstrap(Container))>
        <div
          className="shop_now_panel"
        >
          <div
            className="row product_panel"
          >
            <div
              className="col-md-4 product_img"
            >
              <div
                className="sale"
              >
                SALE
              </div>
              <img
                alt="bag.png"
                src="bag.png"
              />
            </div>
            <div
              className="col-md-8 shop_now"
            >
              <h2>
                Vera Bradley
              </h2>
              <p>
                Carry the day in the style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring colorful faux leather trim,this tote offers a roomy interior plus just enough perfectly placed.
              </p>
              <p>
                <a
                  className="btn btn-lg"
                  href="/categories"
                >
                  Shop Now
                </a>
              </p>
            </div>
          </div>
        </div>
        <div
          className="register_panel"
        >
          <div
            className="row shop_now_panel"
          >
            <div
              className="col-md-4"
            >
              <div
                className="wow_block"
              >
                <h1>
                  WOW
                </h1>
                <h2
                  className="red"
                >
                  Check
                </h2>
                <h2
                  className="red"
                >
                  WHAT!
                </h2>
              </div>
              <div
                className="wow_men_block"
              >
                <h1>
                  Men
                </h1>
              </div>
            </div>
            <div
              className="col-md-8 "
            >
              <div
                className="game_begin_block"
              >
                <img
                  alt="pop_image.png"
                  src="pop_image.png"
                />
                <div
                  className="game_sub_block"
                >
                  <h1>
                    Let The Game begin
                  </h1>
                  <h2>
                    Registration is on - get ready for the Open
                  </h2>
                  <p>
                    <a
                      className="btn btn-lg"
                      href="/login"
                    >
                      Register
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </ForwardRef(Bootstrap(Container))>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_didScheduleRenderPhaseUpdate": false,
    "_dispatcher": Object {
      "readContext": [Function],
      "useCallback": [Function],
      "useContext": [Function],
      "useDebugValue": [Function],
      "useEffect": [Function],
      "useImperativeHandle": [Function],
      "useLayoutEffect": [Function],
      "useMemo": [Function],
      "useReducer": [Function],
      "useRef": [Function],
      "useState": [Function],
    },
    "_element": <Home />,
    "_firstWorkInProgressHook": null,
    "_forcedUpdate": false,
    "_instance": Home {
      "context": Object {},
      "props": Object {},
      "refs": Object {},
      "state": null,
      "updater": Updater {
        "_callbacks": Array [],
        "_renderer": [Circular],
      },
    },
    "_isReRender": false,
    "_newState": null,
    "_numberOfReRenders": 0,
    "_renderPhaseUpdates": null,
    "_rendered": <div
      className="home"
    >
      <div
        className="row header_panel"
      >
        <ForwardRef(Bootstrap(Container))>
          <div
            className="col-md-6"
          >
            <h1>
              Background and development
            </h1>
            <h2>
              Convergent the dictates of the consumer: background and development
            </h2>
            <p>
              <LinkContainer
                activeClassName="active"
                className="btn btn-lg"
                exact={false}
                replace={false}
                strict={false}
                to="/categories"
              >
                <a
                  href="/"
                >
                  View All
                </a>
              </LinkContainer>
            </p>
          </div>
        </ForwardRef(Bootstrap(Container))>
      </div>
      <ForwardRef(Bootstrap(Container))>
        <div
          className="shop_now_panel"
        >
          <div
            className="row product_panel"
          >
            <div
              className="col-md-4 product_img"
            >
              <div
                className="sale"
              >
                SALE
              </div>
              <img
                alt="bag.png"
                src="bag.png"
              />
            </div>
            <div
              className="col-md-8 shop_now"
            >
              <h2>
                Vera Bradley
              </h2>
              <p>
                Carry the day in the style with this extra-large tote crafted in our chic B.B. Collection textured PVC. Featuring colorful faux leather trim,this tote offers a roomy interior plus just enough perfectly placed.
              </p>
              <p>
                <a
                  className="btn btn-lg"
                  href="/categories"
                >
                  Shop Now
                </a>
              </p>
            </div>
          </div>
        </div>
        <div
          className="register_panel"
        >
          <div
            className="row shop_now_panel"
          >
            <div
              className="col-md-4"
            >
              <div
                className="wow_block"
              >
                <h1>
                  WOW
                </h1>
                <h2
                  className="red"
                >
                  Check
                </h2>
                <h2
                  className="red"
                >
                  WHAT!
                </h2>
              </div>
              <div
                className="wow_men_block"
              >
                <h1>
                  Men
                </h1>
              </div>
            </div>
            <div
              className="col-md-8 "
            >
              <div
                className="game_begin_block"
              >
                <img
                  alt="pop_image.png"
                  src="pop_image.png"
                />
                <div
                  className="game_sub_block"
                >
                  <h1>
                    Let The Game begin
                  </h1>
                  <h2>
                    Registration is on - get ready for the Open
                  </h2>
                  <p>
                    <a
                      className="btn btn-lg"
                      href="/login"
                    >
                      Register
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </ForwardRef(Bootstrap(Container))>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_callbacks": Array [],
      "_renderer": [Circular],
    },
    "_workInProgressHook": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home />,
}
`;
